File: homework/homework1/code/p1_count_gray_num.m
% count the number of each pixel value
function gray_num = p1_count_gray_num(origin_grain)
    [w, h] = size(origin_grain);

    % count the number of each pixel value
    gray_num = zeros(1, 256);
    for i = 1:w
        for j = 1:h
            gray_num(origin_grain(i, j) + 1) = gray_num(origin_grain(i, j) + 1) + 1;
        end
    end
end

File: homework/homework1/code/p1/p1_count_gray_num.py
from typing import List
import numpy as np

# count the number of each pixel value
def count_gray_num(origin_image: np.array) -> np.array:
    gray_num = np.zeros(256, dtype=float)
    w, h = origin_image.shape

    for i in range(w):
        for j in range(h):
            gray_num[int(origin_image[i][j])] += 1
    
    return gray_num